server/server/src/payment_service.rs

```diff
@@ -0,0 +1,333 @@
+use crate::file_index_repo::{GetStripeCustomerIdError, FREE_TIER_SIZE};
+use crate::ServerError::{ClientError, InternalError};
+use crate::{file_index_repo, RequestContext, ServerError};
+use lockbook_models::api::{AccountTier, CreditCardInfo, GetRegisteredCreditCardsError, GetRegisteredCreditCardsResponse, RegisterCreditCardError, RegisterCreditCardRequest, RegisterCreditCardResponse, RemoveCreditCardError, RemoveCreditCardRequest, RemoveCreditCardResponse, SwitchAccountTierError, SwitchAccountTierRequest, SwitchAccountTierResponse};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+static STRIPE_ENDPOINT: &str = "https://api.stripe.com/v1";
+static PAYMENT_METHODS_ENDPOINT: &str = "/payment_methods";
+static DETACH_ENDPOINT: &str = "/detach";
+static ATTACH_ENDPOINT: &str = "/attach";
+static CUSTOMER_ENDPOINT: &str = "/customers";
+static SUBSCRIPTIONS_ENDPOINT: &str = "/subscriptions";
+static SETUP_INTENTS_ENDPOINT: &str = "/setup_intents";
+
+#[derive(Serialize, Deserialize)]
+struct BasicStripeResponse {
+    id: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct SetupIntentStripeResponse {
+    status: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct PaymentMethodStripeResponse {
+    id: String,
+    card: PaymentMethodCard,
+}
+
+#[derive(Serialize, Deserialize)]
+struct PaymentMethodCard {
+    last4: String,
+}
+
+pub async fn register_credit_card(
```

Comment:
+ Stripe is gonna send us errors that we want to parse for the credit card, not sure if just a TODO at this point, but you're going to want to model things like `CardNumberNotValid` and other errors that stripe docs suggests it may throw back at us.
+ This fn is ripe for being broken up into smaller routines, this top level `register_credit_card` fn should read like a story about what we do when a someone tries to add a card.
+ If any of these things fail, we're going to roll back our transaction (things won't be saved in our DB), but we're going to create a bunch of junk in stripe. If I'm understanding correctly, if someone messes up their payment info, we're going going to have a customer object for every failure (in stripe). This problem is inherent to trying to reconcile multiple stores of data (we have this sort of problem across s3 & postgres for instance). But there are ways you can greatly decrease this by paying special attention to the order of events.